xaitk_saliency_demo/app/engine/ml_xai.py

```diff
@@ -81,20 +89,41 @@ def classify_images(self, image_iter):
             yield dict(zip(self.get_labels(), out[self.idx]))
 
     def get_config(self):
-        # Required by a parent class.
+        # Required by a parent class. Will not be used in this context.
+        return {}
+
+
+# SMQTK black-box descriptor generator
+class DescrModel(ImageDescriptorGenerator):
+
+    def __init__(self, model):
+        self.model = model
+
+    @torch.no_grad()
+    def generate_arrays_from_images(self, img_mat_iter: Iterable[np.ndarray]) -> Iterable[np.ndarray]:
+        for img in img_mat_iter:
+            inp = imagenet_model_loader(img).unsqueeze(0).to(self.model.device)
+            vec = self.model(inp).cpu().numpy().squeeze()
+            yield vec
+
+    def get_config(self) -> Dict[str, Any]:
+        # Required by a parent class. Will not be used in this context.
         return {}
 
 
+# -----------------------------------------------------------------------------
 class Saliency:
     def __init__(self, model, name, params):
         self._model = model
         try:
-            for key, value in SALIENCY_TYPES[name].items():
-                constructor = value.get("class")
-                param_keys = value.get("params", params.keys())
-                setattr(self, key, constructor(**{k: params[k] for k in param_keys}))
-        except:
+            kv_pairs = SALIENCY_TYPES[name].items()
+        except IndexError:
             logger.info(f"Could not find {name} in {list(SALIENCY_TYPES.keys())}")
+            return
+        for key, value in kv_pairs:
+            constructor = value.get("class")
+            param_keys = value.get("params", params.keys())
```

Comment:
If you don't have any "params" anymore, I'm not sure what you will get here in constructor kwargs.

Reply:
Indeed, you are right, I forgot about those.